schema/schema.py

```diff
@@ -22,20 +30,22 @@ def format_param(param: click.core.Option) -> Dict[str, Union[Optional[str], Lis
                        a list of those allowed options
     * default - The default option for that parameter
     """
-    formatted_param: Dict[str, Union[Optional[str], List[str]]] = {"name": param.name, "help": param.help}
+    formatted_param: Dict[str, Any] = {"title": param.name, "description": param.help}
 
     # NOTE: Params do not have explicit "string" type; either "text" or "path".
     #       All choice options are from a set of strings.
-    if param.type.name in ["text", "path", "choice"]:
+    if param.type.name in ["text", "path", "choice", "filename", "directory"]:
         formatted_param["type"] = "string"
+    elif param.type.name == "LIST":
+        formatted_param["type"] = "array"
     else:
         formatted_param["type"] = param.type.name
```

Comment:
Should we be worried about casing here? Just commenting since the list check above looks for all uppercase.

Reply:
We should, yes. When I went looking through the parameter types to make sure everything was accurate, it was returned exactly like that. I thought it weird as well, since everything else is lowercase, but it's the only other instance that shows up.
![image](https://github.com/aws/aws-sam-cli/assets/51037424/7b9f2f79-1d9b-476e-9e68-e1a444cd0e72)

Reply:
Since that's the case, we should consider converting the parameter type names to lowercase (or uppercase) before doing a comparison against values we check in case anything changes in the future (`LIST` gets updated to `list`)

Reply:
Yup that sounds reasonable, will add those changes